test/Microsoft.ML.AutoML.Tests/DatasetUtil.cs

```diff
@@ -82,5 +84,124 @@ private static string DownloadIfNotExists(string baseGitPath, string dataFile)
 
             throw new Exception($"Failed to download test file {dataFile}.");
         }
+
+        public static string GetFlowersDataset()
+        {
+            const string datasetName = @"flowers";
+            string assetsRelativePath = @"assets";
+            string assetsPath = GetAbsolutePath(assetsRelativePath);
+            string imagesDownloadFolderPath = Path.Combine(assetsPath, "inputs",
+                "images");
+
+            //Download the image set and unzip
+            string finalImagesFolderName = DownloadImageSet(
+                imagesDownloadFolderPath);
+
+            string fullImagesetFolderPath = Path.Combine(
+                imagesDownloadFolderPath, finalImagesFolderName);
+
+            var images = LoadImagesFromDirectory(
+                folder: fullImagesetFolderPath, useFolderNameAsLabel: true);
+
+            using (StreamWriter file = new StreamWriter(datasetName))
+            {
+                file.WriteLine("Label,ImagePath");
+                foreach (var image in images)
+                    file.WriteLine(image.Label + "," + image.ImagePath);
+            }
+
+            return datasetName;
+        }
+
+        public static IEnumerable<ImageData> LoadImagesFromDirectory(string folder,
+            bool useFolderNameAsLabel)
+        {
+            var files = Directory.GetFiles(folder, "*",
+                searchOption: SearchOption.AllDirectories);
+            foreach (var file in files)
+            {
+                if (Path.GetExtension(file) != ".jpg")
+                    continue;
+
+                var label = Path.GetFileName(file);
+                if (useFolderNameAsLabel)
+                    label = Directory.GetParent(file).Name;
+                else
+                {
+                    for (int index = 0; index < label.Length; index++)
+                    {
+                        if (!char.IsLetter(label[index]))
+                        {
+                            label = label.Substring(0, index);
+                            break;
+                        }
+                    }
+                }
+
+                yield return new ImageData()
+                {
+                    ImagePath = file,
+                    Label = label
+                };
+
+            }
+        }
+
+        public static string DownloadImageSet(string imagesDownloadFolder)
+        {
+            string fileName = "flower_photos_tiny_set_for_unit_tests.zip";
+            string url = $"https://mlnetfilestorage.file.core.windows.net/imagesets" +
+                $"/flower_images/flower_photos_tiny_set_for_unit_tests.zip?st=2019" +
+                $"-08-29T00%3A07%3A21Z&se=2030-08-30T00%3A07%3A00Z&sp=rl&sv=2018" +
+                $"-03-28&sr=f&sig=N8HbLziTcT61kstprNLmn%2BDC0JoMrNwo6yRWb3hLLag%3D";
```

Comment:
Recommend linking to aka.ms links, which point to the CDN, which point to public folders in the blobstore.

Linking to aka.ms allows us to redirect the links in the future incase we decide to move them (without breaking old versions. Linking to the CDN gives us speed, and hides the actual blobstore backing the CDN.

As above, Yael's downloader class will help with the downloading process.

Reply:
thanks and I agree we should make short links but because I'm short on time and this is not a pressing change in a test code so I can open an issue and resolve this later.

---
In reply to: [339330533](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339330533) [](ancestors = 339330533)

Reply:
Ok. Great. Can you make a tracking issue for these?

Reply:
yep

---
In reply to: [339927017](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339927017) [](ancestors = 339927017)